transforms/universal/merge/dpk_merge/transform.py

```diff
@@ -0,0 +1,138 @@
+from pathlib import Path
+from typing import Any
+
+import pyarrow as pa
+from data_processing.transform import AbstractTableTransform, TransformConfiguration
+from data_processing.utils import CLIArgumentProvider
+
+
+short_name = "merge"
+cli_prefix = f"{short_name}_"
+input_dirs_key = "input_dirs"
+input_dirs_cli_param = f"{cli_prefix}{input_dirs_key}"
+
+
+class MergeTransform(AbstractTableTransform):
+    """
+    Implements a simple copy of a pyarrow Table.
+    """
+
+    def __init__(self, config: dict[str, Any]):
+        """
+        Initialize based on the dictionary of configuration information.
+        This is generally called with configuration parsed from the CLI arguments defined
+        by the companion runtime, MergeTransformRuntime.  If running inside the RayMutatingDriver,
+        these will be provided by that class with help from the RayMutatingDriver.
+        """
+        # Make sure that the param name corresponds to the name used in apply_input_params method
+        # of MergeTransformConfiguration class
+        super().__init__(config)
+        input_dirs = config.get(input_dirs_cli_param)
+        self.input_dirs = []
+        if input_dirs == None:
+            # For backwards compatibility after we started encouraging the use of cli prefix in the configuration keys
+            input_dirs = config.get(input_dirs_key)
+        if input_dirs is not None:
+            input_dirs = input_dirs.split(",")
+            for input_dir in input_dirs:
+                self.input_dirs.append(Path(input_dir))
+
+    def transform(self, table: pa.Table, file_name: str = None) -> tuple[list[pa.Table], dict[str, Any]]:
+
+        self.logger.debug(f"Transforming one table with {len(table)} rows")
+        file_path = Path(file_name)
+        data_access = self.config.get("data_access")
+        parent_path = Path(data_access.get_input_folder())
+        relative_path = file_path.relative_to(parent_path)
+        added_columns = 0
+        for input_path in self.input_dirs:
+            merged_file_path = input_path / relative_path
+            self.logger.debug(f"merging a table from {merged_file_path}")
+            t, _ = data_access.get_table(str(merged_file_path))
+            table, i = MergeTransform._copy_columns(table, t)
+            added_columns += i
+        # transfer table column names into a set, breaks their order. In order to provide reproducible results we sort
+        # the result table columns.
+        # Sort columns alphabetically
+        sorted_columns = sorted(table.column_names)
+        table = table.select(sorted_columns)
+
+        # Add metadata.
+        self.logger.debug(f"Transformed one table with {len(table)} rows, added {added_columns} columns from {len(self.input_dirs)} tables")
+        metadata = {
+            "nfiles": 1,
+            "nrows": len(table),
+            "merged_tables": len(self.input_dirs),
+            "added_columns": added_columns,
+        }
```

Comment:
I do not know how useful the above 2. Merged tables is len of input dirs * nfiles